common/cmake_modules/GncAddTest.cmake

```diff
@@ -117,27 +117,35 @@ function(gnc_gtest_configure)
   endif()
   unset(GTEST_SRC_DIR CACHE)
   if (GTEST_ROOT)
+    find_path(GTEST_SRC_DIR src/gtest-all.cc NO_CMAKE_SYSTEM_PATH
+      PATHS ${GTEST_ROOT} PATH_SUFFIXES googletest)
+  elseif (GMOCK_ROOT)
+    find_path(GTEST_SRC_DIR src/gtest-all.cc NO_CMAKE_SYSTEM_PATH
+      PATHS ${GMOCK_ROOT}/gtest)
```

Comment:
This handles the googletest v1.7 case but not v 1.8+, where we might have `${GMOCK_ROOT}/googletest/src/gtest_all.cc` or `${GMOCK_ROOT}/../googletest/src/gtest_all.cc`.

Reply:
Do you really want to allow both GTEST_ROOT and GMOCK_ROOT as root directory for v1.8+, i.e. should they really be equivalent?
Does this make sense? Since v1.8+ the name of the repository is googletest. From my point of view GTEST_ROOT is sufficient for future versions and GMOCK_ROOT is deprecated since v1.8+.

Reply:
My reasoning is that the user may have a script written (or a habit formed) for 1.7 where they set GMOCK_ROOT only. Why break it when it's so simple not to? I suppose the counter argument is that a user might want to use the distribution-provided gtest but a separate gmock by setting GMOCK_ROOT and they'd be frustrated because having GMOCK_ROOT set would either find both sets of sources or wouldn't find the googletest sources and so fail.

ISTM the second scenario is less likely.

Reply:
At first to avoid misunderstandings, I already considered the case `${GMOCK_ROOT}/../googletest/src/gtest_all.cc` in my last proposal. My question above is only related to the case `${GMOCK_ROOT}/googletest/src/gtest_all.cc`, where `GMOCK_ROOT` is used as the GoogleTest v1.8+ root directory. Maybe this already clarifies your last remark.

>  My reasoning is that the user may have a script written (or a habit formed) for 1.7 where they set GMOCK_ROOT only. Why break it when it's so simple not to? I suppose the counter argument is that a user might want to use the distribution-provided gtest but a separate gmock by setting GMOCK_ROOT

I don't know exactly what case do have in mind. Why should something get broken?

Reply:
Suppose a packager's distro currently provides googlemock 1.7 from the googletest distribution tarball as their gmock source package and so they've written their gnucash build script to pass GMOCK_ROOT=somewhere/gmock and are relying on GnuCash finding the google*test* sources as a subdir. Now their next release is going to have googletest 1.8 with both googletest and googlemock subdirs. It would be kind to not require them to change their GnuCash buildscript, especially in 3.7/3.8. *But* if we decide to apply this to GnuCash 4 (i.e.master), require googletest 1.8 or later , and remove GMOCK_ROOT entirely then it's reasonable to expect packagers to change their scripts as necessary.

Reply:
If only `GMOCK_ROOT` is defined, but not `GTEST_ROOT`, even on current `maint` branch it's not possible that GnuCash finds the GoogleTest sources, when switching to v1.8.
Therefore I don't exactly understand, where this requirement comes from.